v1/s012-pyglet/pong.html

```diff
@@ -0,0 +1,616 @@
+                        Hra bude pro 2 hráče, nebudeme programovat chování počítače.
+                        Každý z hráčů může ovládat svou pálku stiskem kláves.
+                        Jeden hráč šipkou nahoru a šipkou dolů a druhý hráč klávesami
+                        W a S.
+
+                        Stav hry dokážeme v Pythonu vyjádřit pomocí proměnných
+                        a konstant. To dává smysl, protože některé
+                        věci se ve hře mění (poloha pálek, poloha míčku, rychlost míčku, skóre)
+                        a některé ne (velikost hrací plochy, velikost pálek a
+                        míčku, pálek a míčku, poloha a velikost počítadel skóre). 
+                        Ze složitějších datových struktur použijeme seznam
+                        (list, ten už známe) a množinu (set), která je velmi podobná
+                        matematické množině. Zjednodušeně je to seznam, který se
+                        nestará o pořadí a stejné prvky v něm mohou být právě jednou.
+                    </div>
+
+                    <div>
+                        Založte si nový skript a zkuste v Pythonu nadefinovat
+                        to jednodušší - <strong>konstanty</strong>, které 
+                        budeme v tvorbě hry potřebovat. Jen připomeňme, že
+                        názvy konstant se v Pythonu píší velkými písmeny.
+<details class="solution notes"><h4>Řešení</h4>
+<pre>
+# Velikost okna (v pixelech)
+SIRKA = 900
+VYSKA = 600
+
+VELIKOST_MICE = 20
+TLOUSTKA_PALKY = 10
+DELKA_PALKY = 100
+RYCHLOST = 200  # v pixelech za sekundu
+RYCHLOST_PALKY = RYCHLOST * 1.5  # taky v pixelech za sekundu
+
+DELKA_PULICI_CARKY = 20
+VELIKOST_FONTU = 42
+ODSAZENI_TEXTU = 30
+</pre>
+</details>
+
+                    </div>
+                    <div>
+                        Zkuste se znovu zamyslet a nadefinovat <strong>proměnné</strong>
+                        potřebné v naší hře jako je poloha míčku, poloha pálek,
+                        rychlost míče, stisknuté klávesy nebo skóre.
+                        Dejte jim hodnoty, které by měly mít na začátku hry.
+                        Která proměnná bude číslo? Která bude seznam nebo
+                        množina? Kterou proměnnou na začátku vypočítáme
+                        z hodnoty nějaké konstanty?
+                        Budou to globální proměnné, za což by nás profesionální
+                        programátor pokáral, ale nám to v tuhle chvíli ulehčí
+                        práci.
+<details class="solution notes"><h4>Řešení</h4>
+<pre>
+pozice_palek = [VYSKA // 2, VYSKA // 2]  # vertikalni pozice dvou palek
+pozice_mice = [0, 0]  # x, y souradnice micku -- nastavene v reset()
+rychlost_mice = [0, 0]  # x, y slozky rychlosti micku -- nastavene v reset()
+stisknute_klavesy = set()  # sada stisknutych klaves
+skore = [0, 0]  # skore dvou hracu
+</pre>
+</details>
+                    </div>
+                </section>                
```

Comment:
Bylo super vysvětlit `event loop` a `event-driven programming` (a vztáhnout to k dřívějším hrám) ať tam je trochu teoretický základ.
Koukni se na http://pyladies.cz/v1/s012-pyglet/pyglet.html, sekce "Interaktivní programy".